js/setpassword.js

```diff
@@ -0,0 +1,69 @@
+SetPassword = {
+	init : function() {
+		$('#set-password #submit').click(SetPassword.onClickSetPassword);
+	},
+
+	onClickSetPassword : function(event){
+		event.preventDefault();
+		var passwordObj = $('#password');
+		if (passwordObj.val()){
+			$.post(
+				passwordObj.parents('form').attr('action'),
+				{password : passwordObj.val()}
+			).done(function (result) {
+				SetPassword._resetDone(result);
+			}).fail(function (result) {
+				SetPassword._onSetPasswordFail(result);
+			});
+		}
+	},
+
+	_onSetPasswordFail: function(result) {
+		var responseObj = JSON.parse(result.responseText);
+		var errorObject = $('#error-message');
+		var showErrorMessage = false;
+
+		if (responseObj.type === 'usererror') {
+			errorObject.text(responseObj.message);
+			showErrorMessage = true;
+		} else if (responseObj.type === 'emailsendfailed') {
+			errorObject.text(responseObj.message);
```

Comment:
a simpler pattern would be to set a message to a variable "errorMessage  = ..." and have it null by default.
Then at the bottom your if becomes `if (errorMessage) {` then set it and show it.

(optional change)

Reply:
Updated.